### DISCUSSION TITLE
how did you make that emplacement gun tau cannon?

### DISCUSSION THREAD

**Discussion by luhohagaming** [Pinsplash/halflife2chaos#113] [General]
i really wanna try it in vanilla hl2 but i need to know how you made it, and if you can, plz help me implement it into my hl2 :p

**Comment by Pinsplash** [comment]
there isn't a tau cannon one. i think you mean the autogun one from episode two, but regardless they're all func_tank entities like func_tank_combine_cannon. you shouldn't need to do any coding to make them work, at least in ep2.

**Reply by luhohagaming** [reply]
lol i meant the airboat pulse cannon one

**Reply by luhohagaming** [reply]
is there like a command to spawn them? what do i do to spawn it

**Reply by luhohagaming** [reply]
im talking about hl2 base game btw not the episodes

**Reply by luhohagaming** [reply]
also, last thing. ent_create and give func_tank_combine_cannon makes just a laser tho. im probably just being stupid. but this is unrelated how do i spawn the functioning airboat pulse cannon emplacement gun?

**Reply by Pinsplash** [reply]
the entity for it is func_tankairboatgun, but you're not meant to spawn it with ent_create so it's not going to work right, same as func_tank_combine_cannon. i had to edit the code to some degree just to make the entity controllable without forcing the player to stand inside a trigger, (among some other stuff i think) so yeah if you're wanting to spawn them *not* from hammer it's a lot more complicated.

**Reply by luhohagaming** [reply]
so how can i spawn it in HL2 chaos?

**Reply by luhohagaming** [reply]
how do i spawn it in hl2 chaos?

**Reply by Pinsplash** [reply]
oh, hi. first set chaos_rng1 to 1 and then do chaos_test_effect 54.

**Reply by luhohagaming** [reply]
ok last question what are the chaos_Test_effect ids?

**Reply by Pinsplash** [reply]
you can see them all with chaos_print
